opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -213,10 +213,22 @@ class EclMaterialLawManager
         return *materialLawParams_[elemIdx];
     }
 
-    std::shared_ptr<MaterialLawParams>& materialLawParamsPointerReferenceHack(unsigned elemIdx)
+    MaterialLawParams& materialLawParamsWellConnection(unsigned elemIdx)
     {
         assert(0 <= elemIdx && elemIdx <  materialLawParams_.size());
-        return materialLawParams_[elemIdx];
+        return *materialLawParams_[elemIdx];
+    }
+
+    const MaterialLawParams& materialLawParamsWellConnection(unsigned int connectionIdx) const
```

Comment:
How to make sure that the connectionIdx maps to the right well Connection Parameter?